spec/models/mixins/miq_web_server_worker_mixin_spec.rb

```diff
@@ -68,6 +68,22 @@ def self.binding_address
       end
     end
 
-    expect(test_class.binding_address).to eq (before + "JUNK")
+    expect(test_class.binding_address).to eq(before + "JUNK")
+  end
+
+  describe '#rails_application' do
+    context 'websocket' do
+      let(:worker) { FactoryGirl.create(:miq_websocket_worker, :uri => 'http://127.0.0.1:3000') }
+      it 'creates a websocket server' do
+        expect(worker.rails_application).to be_a_kind_of(WebsocketServer)
+      end
+    end
+
+    context 'rails' do
+      let(:worker) { FactoryGirl.create(:miq_ui_worker, :uri => 'http://127.0.0.1:3000') }
+      it 'creates a rails server' do
+        expect(worker.rails_application).to be_a_kind_of(Vmdb::Application)
```

Comment:
Can you collapse these two tests into one where you have two different workers instantiated but having different rails_application values?  I think it's important to demonstrate that the ivar is NOT shared with each worker instance.